src/main/kotlin/org/sol4k/instruction/SplTransferInstruction.kt

```diff
@@ -1,27 +1,40 @@
 package org.sol4k.instruction
 
+import okio.Buffer
 import org.sol4k.AccountMeta
-import org.sol4k.Binary
 import org.sol4k.Constants.TOKEN_PROGRAM_ID
 import org.sol4k.PublicKey
 
 class SplTransferInstruction(
     from: PublicKey,
     to: PublicKey,
+    mint: PublicKey,
     owner: PublicKey,
+    signers: List<PublicKey>,
     private val amount: Long,
+    private val decimals: Int,
 ) : Instruction {
+    companion object {
+        @Suppress("unused")
+        private const val instructionTransfer = 3
```

Comment:
Use `TransferChecked` instead of `Transfer` for SPL tokens here to check token mint and decimals.